### PR TITLE
Replace ioutil with os and io for the test

### LINKED ISSUE #100367: Fixes for Deprecation of io/ioutil (in Go 1.16)
Please see https://golang.org/doc/go1.16#ioutil

We have a lot of references to `ioutil` which we need to migrate to newer package(s) mentioned in the url above:
https://cs.k8s.io/?q=io%2Fioutil&i=nope&files=.*%5C.go&excludeFiles=&repos=kubernetes/kubernetes

We also need some verify scripts to ensure we don't add back this package

/area code-organization

### PR DESCRIPTION
**Pull request by ahrtr**
The ioutil package has already been deprecated in golang 1.16, please see [go1.16#ioutil](https://golang.org/doc/go1.16#ioutil). So we need to replace all the ioutil with os and io.

This PR is just to update the test. There will be a couple of separate PRs for another SIGs. 

Previously there is a [PR](https://github.com/kubernetes/kubernetes/pull/105994), which is huge. So I am splitting the huge PR into a couple of smaller PRs. This PR is just one of them. 

Fix https://github.com/kubernetes/kubernetes/issues/100367

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
Tests are not particularly affected by this comment: https://github.com/kubernetes/kubernetes/issues/100367#issuecomment-802591356 so it is probably all right to switch.

/priority backlog
/triage accepted

The PR is huge so already has merge conflicts. Some synchronization will be needed between getting approvals first, then rebasing and lgtm right away.

/approve

for node

**Comment by ahrtr** [comment]
> Tests are not particularly affected by this comment: [#100367 (comment)](https://github.com/kubernetes/kubernetes/issues/100367#issuecomment-802591356) so it is probably all right to switch.
> 
> /priority backlog /triage accepted
> 
> The PR is huge so already has merge conflicts. Some synchronization will be needed between getting approvals first, then rebasing and lgtm right away.
> 
> /approve
> 
> for node

Thanks for the comment. Rebasing done!

File: test/images/apparmor-loader/loader.go [hunk 1]
```diff
@@ -118,7 +117,7 @@ func loadNewProfiles() (success bool, newProfiles []string) {
 
 	success = true
 	for _, dir := range dirs {
-		infos, err := ioutil.ReadDir(dir)
+		infos, err := os.ReadDir(dir)
```
Comment (spiffxp) [inline test/images/apparmor-loader/loader.go:120]: same thing, you need to update `test/images/apparmor-loader/VERSION`

File: test/images/metadata-concealment/check_metadata_concealment.go [hunk 2]
```diff
@@ -128,7 +128,7 @@ func checkURL(url string, header http.Header, expectedStatus int, expectedToCont
 	if resp.StatusCode != expectedStatus {
 		return fmt.Errorf("unexpected response: got %d, want %d", resp.StatusCode, expectedStatus)
 	}
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
```
Comment (spiffxp) [inline test/images/metadata-concealment/check_metadata_concealment.go:131]: same thing, you need to update `test/images/metadata-concealment/VERSION`

File: test/images/pets/peer-finder/peer-finder.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func shellOut(sendStdin, script string) {
 	stdin.Write([]byte(sendStdin))
 	stdin.Close()
 
-	out, err := ioutil.ReadAll(stdout)
+	out, err := io.ReadAll(stdout)
```
Comment (spiffxp) [inline test/images/pets/peer-finder/peer-finder.go:80]: same thing, you need to update `test/images/peer-finder/VERSION`

**Review by spiffxp — CHANGES_REQUESTED** [review]
I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

**Comment by ahrtr** [comment]
> I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

Thanks for the comment. Removed all changes in test/images/*, and rebased the PR.

**Comment by ahrtr** [comment]
Rebased.

**Comment by ahrtr** [comment]
> I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

@spiffxp  Just submitted a separate PR [pull/106415](https://github.com/kubernetes/kubernetes/pull/106415 )

**Review by ehashman — COMMENTED** [review]
/approve

for 
test/e2e/instrumentation/monitoring/stackdriver_metadata_agent.go
test/instrumentation/main.go

**Comment by marosset** [comment]
/remove-sig windows
Ping me or add sig-windows back if there are any concerns specific to Windows.

**Comment by ahrtr** [comment]
> @ahrtr: PR needs rebase.

Just rebased.

**Comment by dims** [comment]
/assign @MadhavJivrajani 

can you please review this Madhav? same as the last one. should be mostly mechanical changes

**Review by MadhavJivrajani — CHANGES_REQUESTED** [review]
Assuming changes to `images/` are in a seperate PR, there are a few more instances to cover:

- test/e2e_node/kubeletconfig/kubeletconfig.go
- test/e2e_node/device_manager_test.go
- test/integration/apiserver/openapiv3_test.go
- test/integration/auth/podsecurity_test.go

Please also recheck in any other similar PRs that you may have opened, thanks!

**Comment by ahrtr** [comment]
> Assuming changes to `images/` are in a seperate PR, there are a few more instances to cover:
> 
>     * test/e2e_node/kubeletconfig/kubeletconfig.go
> 
>     * test/e2e_node/device_manager_test.go
> 
>     * test/integration/apiserver/openapiv3_test.go
> 
>     * test/integration/auth/podsecurity_test.go
> 
> 
> Please also recheck in any other similar PRs that you may have opened, thanks!

Thanks @MadhavJivrajani , just rebased this PR and resolved your comment. PTAL, thanks.

**Review by MadhavJivrajani — APPROVED** [review]
LGTM
Thanks!
This is good to go @dims :)

**Pull request merged**